src/mlpack/methods/ann/ffn.hpp

```diff
@@ -172,9 +172,10 @@ class FFN
    * @param i Index of points to use for objective function gradient evaluation.
    * @param gradient Matrix to output gradient into.
    */
+  template<typename eT>
   void Gradient(const arma::mat& parameters,
                 const size_t i,
-                arma::mat& gradient);
+                arma::Mat<eT>& gradient);
```

Comment:
Sure. I will make the RNN class supported as well. Ahh... I think the reason was that Gradient function already exists in that class apart from this function. Thus I had to templatize it to make it work.